lib/Conversion/TosaToXTenNN.cpp

```diff
@@ -0,0 +1,225 @@
+//===- TosaToXTenNN.cpp -----------------------------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2023 Advanced Micro Devices, Inc.
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetail.h"
+#include "mlir/IR/Matchers.h"
+#include "mlir/Transforms/CommutativityUtils.h"
+#include "xten/Conversion/TosaToXTenNNPass.h"
+#include "xten/Dialect/XTenNN/IR/XTenNNOps.h"
+
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+using namespace mlir;
+
+namespace {
+
+/// Checks to see if the \p castOp is a dequantize operation.
+///
+///\param castOp check that the operation casts from i8, assuming it is signed
+/// to f32.
+/// \return true represents a dequantize cast
+/// \return false does not represent a dequantize cast
+bool isI8ToF32Cast(tosa::CastOp castOp) {
+  TensorType inputTensorType =
+      cast<TensorType>(castOp->getOperand(0).getType());
+  TensorType outputTensorType =
+      cast<TensorType>(castOp->getResult(0).getType());
+  return outputTensorType.getElementType().isF32() &&
+         inputTensorType.getElementType().isInteger(8) &&
+         inputTensorType.getElementType().isSignlessInteger();
```

Comment:
Why restrict it to signless here? I this just temporary?

Reply:
Yes simply due to the restriction in TOSA where signless means signed at the moment. I would also love to have `si<>` and `ui<>` in the future. 
For our use case we will restrict ourselves to f32 and i8.